pkg/proxy/routes.go

```diff
@@ -196,3 +207,26 @@ func removeFromProxyRoutesIPv6() error {
 
 	return nil
 }
+
+// removeStaleProxyRulesIPv6 removes stale proxy rules. This is a v1.15 only function.
+func removeStaleProxyRulesIPv6() error {
+	return removeProtoUnspecRules(netlink.FAMILY_V6)
+}
+
+// removeProtoUnspecRules removes all routing rules with protocol RTPROT_UNSPEC. This is a v1.15 only function.
+func removeProtoUnspecRules(family int) error {
+	rules, err := route.ListRules(family, nil)
+	if err != nil {
+		return fmt.Errorf("listing routing rules: %w", err)
+	}
+	for _, r := range rules {
+		if r.Protocol == unix.RTPROT_UNSPEC {
```

Comment:
Should we also ensure that we only remove rules from the tables that concern us?

Reply:
Sure! Now it also checks `r.Table`:
```diff
- if r.Protocol == unix.RTPROT_UNSPEC {
+ if r.Protocol == unix.RTPROT_UNSPEC && slices.Contains(ciliumRouteTables, r.Table) {
```